### ISSUE TITLE
Blacklisting `.bat` extension disables CSV file upload

### ISSUE THREAD

**Issue by anuveyatsu** [data-govt-nz/ckanext-security#42]
When uploading CSV files, guessed mimetype is `text/plain` (see [this line](https://github.com/data-govt-nz/ckanext-security/blob/master/ckanext/security/resource_upload_validator.py#L46)). According to python's `mimetypes` library, `text/plain` mimetype maps to the following standard file extensions:

`'text/plain': ['.ksh', '.pl', '.c', '.bat', '.h', '.txt']`

If you need to blacklist any of these extensions, it would disable uploading CSV files.

**Comment by ThrawnCA** [comment]
The current solution for #41 would be able to handle this. By ensuring that file extensions have to match the sniffed contents, it allows you to whitelist by file extension instead of content type, so you can just not allow `.bat` and it should be reliably blocked.

Currently it doesn't have a blacklist mode, but that shouldn't be hard to add. Just add a `blocked_extensions` key to the config file alongside the `allowed_extensions` key, then do some checking to see which one is present (with the blacklist presumably taking priority if specified).

**Comment by markstuart** [comment]
Hi @anuveyatsu, thanks for the feedback. I'm unsure how you are seeing this behaviour, we are using this blacklist in the data.govt.nz CKAN instance and have no issues with uploading CSV files:
```
ckanext.security.upload_blacklist = ['.js', '.vbs', 'text/x-msdos-batch', '.bat', '.cmd', '.com', 'application/CDFV2', '.msi', '.msp', '.reg', 'text/x-php', '.php']
```
As you can see, the blacklist includes '.bat'

How have you configured your blacklist?

**Comment by markstuart** [comment]
These are the detected extensions/mimetypes I see in the logs when uploading a CSV file:
```
2021-04-14 13:16:31,958 INFO  [ckanext.security.resource_upload_validator] Detected extensions/mimetypes: ['.brf', u'.csv', '.asc', '.pot', '.text', '.txt', 'text/csv', '.srt', '.ksh', 'text/plain']
```

**Comment by ThrawnCA** [comment]
@anuveyatsu Do you want to raise an issue on https://github.com/qld-gov-au/ckanext-resource-type-validation for implementing a blacklist mode?

**Comment by markstuart** [comment]
This is resolved by #61

**Comment by markstuart** [comment]
Fix is available in tag 3.0.3 of this repo
